crates/ide-assists/src/handlers/merge_imports.rs

```diff
@@ -68,24 +68,31 @@ pub(crate) fn merge_imports(acc: &mut Assists, ctx: &AssistContext<'_>) -> Optio
         (selection_range, edits?)
     };
 
+    // FIXME: Is this the best to get a `SyntaxNode` object? We need one for `SourceChangeBuilder::make_editor`.
+    let parent_node = match ctx.covering_element() {
+        SyntaxElement::Node(n) => n,
+        SyntaxElement::Token(t) => t.parent()?,
+    };
+    let make = SyntaxFactory::new();
+
     acc.add(
         AssistId("merge_imports", AssistKind::RefactorRewrite),
         "Merge imports",
         target,
         |builder| {
-            let edits_mut: Vec<Edit> = edits
-                .into_iter()
-                .map(|it| match it {
-                    Remove(Either::Left(it)) => Remove(Either::Left(builder.make_mut(it))),
-                    Remove(Either::Right(it)) => Remove(Either::Right(builder.make_mut(it))),
-                    Replace(old, new) => Replace(builder.make_syntax_mut(old), new),
-                })
-                .collect();
-            for edit in edits_mut {
+            let mut editor = builder.make_editor(&parent_node);
+            for edit in edits {
```

Comment:
I think that to avoid the intersection assertion failure, we should use `SyntaxEditor::replace_all` for consecutive text ranges instead of individual `insert` or `delete` calls

Reply:
So the edits are generated in a vector by calling either `use_item.try_merge_from()` or `use_tree.try_merge_from()`. How can we combine all of these edits into a single `replace_all()` call?